packages/aws-rfdk/lib/core/lib/x509-certificate.ts

```diff
@@ -229,6 +273,92 @@ abstract class X509CertificateBase extends Construct {
         StringEquals: tagCondition,
       },
     }));
+
+    this.lambdaMonitor = this.addLambdaMonitorFunction({
+      uniqueValue,
+      lambdaLayers: [openSslLayer],
+      lambdaCode: props.lambdaCode,
+      tagCondition,
+      lambdaMonitorHandler: props.lambdaMonitorHandler,
+    });
+    this.addExpDateValidation({
+      uniqueValue,
+      alarmEmailAddress: props.alarmEmailAddress,
+    });
+  }
+
+  private addLambdaMonitorFunction(props: LambdaMonitorFunctionProps): LambdaFunction{
+    const lambdaMonitor = new LambdaFunction(this, 'MonitoringExpDate', {
+      description: `Used by a X509Certificate ${Names.uniqueId(this)} to monitor certificate expiration date.`,
+      code: props.lambdaCode,
+      environment: {
+        UNIQUEID: props.uniqueValue,
+      },
+      runtime: Runtime.NODEJS_14_X,
+      layers: props.lambdaLayers,
+      handler: props.lambdaMonitorHandler,
+      timeout: Duration.seconds(90),
+      logRetention: RetentionDays.ONE_WEEK,
+    });
+    lambdaMonitor.addToRolePolicy(new PolicyStatement({
+      actions: [
+        'secretsmanager:GetSecretValue',
+      ],
+      resources: ['*'],
+      conditions: {
+        StringEquals: props.tagCondition,
+      },
+    }));
+    lambdaMonitor.addToRolePolicy(new PolicyStatement({
+      actions: [
+        'secretsmanager:ListSecrets',
+      ],
+      resources: ['*'],
+    }));
+    lambdaMonitor.addToRolePolicy(new PolicyStatement({
+      actions: [
+        'cloudwatch:PutMetricData',
+      ],
+      resources: ['*'],
+      conditions: {
+        StringEquals: { 'cloudwatch:namespace': CertificateRotationMonitor.METRIC_NAMESPACE },
+      },
+    }));
+    return lambdaMonitor;
+  }
+
+  private addExpDateValidation(props: X509CertValidationProps){
+    const certMetric = new Metric({
+      metricName: CertificateRotationMonitor.METRIC_NAME,
+      namespace: CertificateRotationMonitor.METRIC_NAMESPACE,
+      dimensions: {},
+      // Minimum data point daily
```

Comment:
This comment doesn't really add any value beyond what the code already says, maybe we could change it to:
```suggestion
      // We're looking at the minimum value reported by the Lambda in a day, which should
      // be the only value, since it runs once daily.
```